pkg/importer/data-processor.go

```diff
@@ -264,16 +264,32 @@ func (dp *DataProcessor) convert(url *url.URL) (ProcessingPhase, error) {
 }
 
 func (dp *DataProcessor) resize() (ProcessingPhase, error) {
-	// Resize only if we have a resize request, and if the image is on a file system pvc.
 	size, _ := getAvailableSpaceBlockFunc(dp.dataFile)
 	klog.V(3).Infof("Available space in dataFile: %d", size)
+	isBlockDev := size >= int64(0)
 	shouldPreallocate := false
-	if dp.requestImageSize != "" && size < int64(0) {
-		var err error
-		klog.V(3).Infoln("Resizing image")
-		shouldPreallocate, err = ResizeImage(dp.dataFile, dp.requestImageSize, dp.availableSpace)
-		if err != nil {
-			return ProcessingPhaseError, errors.Wrap(err, "Resize of image failed")
+	if !isBlockDev {
+		// Unlike block devices, a raw image file on a disk doesn't fill up the
+		// file system space as needed (thus we resize it)
+		// And may be bigger than the available space (sparse).
+		// Make sure that we use the biggest bound possible.
+		if dp.requestImageSize != "" {
+			var err error
+			klog.V(3).Infoln("Resizing image")
+			shouldPreallocate, err = ResizeImage(dp.dataFile, dp.requestImageSize, dp.availableSpace)
+			if err != nil {
+				return ProcessingPhaseError, errors.Wrap(err, "Resize of image failed")
+			}
+		} else {
+			// A sparse file could still be bigger than the available space.
+			dataFileURL, err := url.Parse(dp.dataFile)
```

Comment:
how is the check related to the comment?

Reply:
I am describing what this block does, would it be more clear if I added a space between the code and the comment?

Reply:
"A sparse file could still be bigger than the available space" is describing what this block does?

Reply:
After some looking at this code block, I rewrote it so it requires less comments to explain. Hopefully I didn't uncover some edge case where it fails a test!